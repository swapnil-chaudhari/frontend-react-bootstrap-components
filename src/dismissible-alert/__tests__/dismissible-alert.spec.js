import React from 'react';
import renderShallow from 'render-shallow';
import DismissibleAlert from '../dismissible-alert';
import { findWithType } from 'react-shallow-testutils';

describe('<DismissibleAlert>', () => {
    describe('when rendered with default props', () => {
        let component;
        const props = {
            titleText: 'Do Stuff Please',
            subtitleText: 'Please do things to dismiss',
            onDismiss: () => {}
        };

        beforeAll(() => {
            component = renderShallow(<DismissibleAlert { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with custom props', () => {
        let component;
        const props = {
            style: 'warning',
            titleText: 'Do Stuff Please',
            subtitleText: 'Please do things to dismiss',
            buttonText: 'DISMISS',
            onDismiss: () => {}
        };

        beforeAll(() => {
            component = renderShallow(<DismissibleAlert { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with description', () => {
        let component;
        const props = {
            style: 'warning',
            titleText: 'Do Stuff Please',
            subtitleText: 'Please do things to dismiss',
            buttonText: 'DISMISS',
            description: 'Some description',
            onDismiss: () => {}
        };

        beforeAll(() => {
            component = renderShallow(<DismissibleAlert { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when rendered with style success', () => {
        let component;
        const props = {
            buttonText: 'Discard',
            onDismiss: () => {},
            style: 'success',
            subtitleText: 'Subtitle',
            titleText: 'Hi there'
        };

        beforeAll(() => {
            component = renderShallow(<DismissibleAlert { ...props } />).output;
        });

        test('has expected markup', () => {
            expect(component).toMatchSnapshot();
        });
    });


    describe('when the alert is dismissed', () => {
        const props = {
            titleText: 'Do Stuff Please',
            subtitleText: 'Please do things to dismiss',
            buttonText: 'DISMISS',
            onDismiss: jest.fn()
        };

        beforeAll(() => {
            const component = renderShallow(<DismissibleAlert { ...props } />).output;
            const button = findWithType(component, 'button');
            button.props.onClick();
        });

        test('calls the onDismiss property', () => {
            expect(props.onDismiss).toHaveBeenCalledTimes(1);
        });
    });

    describe('when rendered with a blank button text', () => {
        let component;
        const props = {
            titleText: 'Do Stuff Please',
            subtitleText: 'Please do things to dismiss',
            buttonText: '',
            style: 'danger',
            onDismiss: () => {}
        };

        beforeAll(() => {
            component = renderShallow(<DismissibleAlert { ...props } />).output;
        });

        test('renders the component without the dismiss button', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
