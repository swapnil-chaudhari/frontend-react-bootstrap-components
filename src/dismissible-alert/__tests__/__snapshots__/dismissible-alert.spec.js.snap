// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DismissibleAlert> when rendered with a blank button text renders the component without the dismiss button 1`] = `
<div
  className="DismissibleAlert DismissibleAlert-danger"
>
  <IncompleteAlert />
  <div
    className="DismissibleAlert-danger-titles"
  >
    <h3
      className="DismissibleAlert-danger-title"
    >
      Do Stuff Please
    </h3>
    <p
      className="DismissibleAlert-danger-subtitle"
    >
      Please do things to dismiss
    </p>
  </div>
  
</div>
`;

exports[`<DismissibleAlert> when rendered with custom props has expected markup 1`] = `
<div
  className="DismissibleAlert DismissibleAlert-warning"
>
  <IncompleteAlert />
  <div
    className="DismissibleAlert-warning-titles"
  >
    <h3
      className="DismissibleAlert-warning-title"
    >
      Do Stuff Please
    </h3>
    <p
      className="DismissibleAlert-warning-subtitle"
    >
      Please do things to dismiss
    </p>
  </div>
  <button
    className="DismissibleAlert-warning-button"
    onClick={[Function]}
  >
    DISMISS
  </button>
</div>
`;

exports[`<DismissibleAlert> when rendered with default props has expected markup 1`] = `
<div
  className="DismissibleAlert DismissibleAlert-default"
>
  <IncompleteAlert />
  <div
    className="DismissibleAlert-default-titles"
  >
    <h3
      className="DismissibleAlert-default-title"
    >
      Do Stuff Please
    </h3>
    <p
      className="DismissibleAlert-default-subtitle"
    >
      Please do things to dismiss
    </p>
  </div>
  <button
    className="DismissibleAlert-default-button"
    onClick={[Function]}
  >
    DISMISS
  </button>
</div>
`;

exports[`<DismissibleAlert> when rendered with description has expected markup 1`] = `
<div
  className="DismissibleAlert DismissibleAlert-warning DismissibleAlert-high"
>
  <IncompleteAlert />
  <div
    className="DismissibleAlert-warning-titles"
  >
    <h3
      className="DismissibleAlert-warning-title"
    >
      Do Stuff Please
    </h3>
    <p
      className="DismissibleAlert-warning-subtitle"
    >
      Please do things to dismiss
    </p>
    <div
      className="DismissibleAlert-warning-description"
    >
      Some description
    </div>
  </div>
  <button
    className="DismissibleAlert-warning-button DismissibleAlert-high-button"
    onClick={[Function]}
  >
    DISMISS
  </button>
</div>
`;

exports[`<DismissibleAlert> when rendered with style success has expected markup 1`] = `
<div
  className="DismissibleAlert DismissibleAlert-success"
>
  <CompleteCheckmark />
  <div
    className="DismissibleAlert-success-titles"
  >
    <h3
      className="DismissibleAlert-success-title"
    >
      Hi there
    </h3>
    <p
      className="DismissibleAlert-success-subtitle"
    >
      Subtitle
    </p>
  </div>
  <button
    className="DismissibleAlert-success-button"
    onClick={[Function]}
  >
    Discard
  </button>
</div>
`;
